modules/internal/ChapelIteratorSupport.chpl

```diff
@@ -17,15 +17,35 @@
  * limitations under the License.
  */
 
+// This comment is a lie for chpl-docs sake. It only applies to the
+// vectorizeOnly iterators found at the bottom of this file.
+/*
+  Data parallel constructs (such as ``forall`` loops) are implicitly
+  vectorizable. If the ``--vectorize`` compiler flag is thrown (implied by
+  ``--fast``), the Chapel compiler will emit vectorization hints to the backend
+  compiler.
```

Comment:
In the man page I said "If enabled, hints will always be generated, but the effects will vary based on the target compiler." as a copout to having to maintain which compilers are actually affected. Does doing the same here sound ok?